redis/db/redisbp/monitored_client.go

```diff
@@ -72,6 +83,75 @@ func (cc *ClusterClient) Wait(ctx context.Context, args WaitArgs) (replicas int6
 	return
 }
 
+var (
+	gaugeLabels = []string{
+		nameLabel,
+	}
+
+	totalConnectionsDesc = prometheus.NewDesc(
+		prometheus.BuildFQName(promNamespace, subsystemPool, "connections"),
+		"Number of total connections in this redisbp pool",
+		gaugeLabels,
+		nil,
+	)
+
+	idleConnectionsDesc = prometheus.NewDesc(
+		prometheus.BuildFQName(promNamespace, subsystemPool, "idle_connections"),
+		"Number of idle connections in this redisbp pool",
+		gaugeLabels,
+		nil,
+	)
+
+	staleConnectionsDesc = prometheus.NewDesc(
+		prometheus.BuildFQName(promNamespace, subsystemPool, "stale_connections"),
+		"Number of stale connections in this redisbp pool",
+		gaugeLabels,
+		nil,
+	)
+)
+
+// Exporter provides an interface for Prometheus merics.
+type Exporter struct {
+	client PoolStatser
+	name   string
+}
+
+func NewExporter(client PoolStatser, name string) *Exporter {
+	return &Exporter{
+		client: client,
+		name:   name,
+	}
+}
+
+// Describe implements the prometheus.Collector interface.
+func (e *Exporter) Describe(ch chan<- *prometheus.Desc) {
+	// All metrics are described dynamically.
```

Comment:
Is this necessary though, wouldn't we be able to send all descriptors in this instance here?
```
  // The sent descriptors fulfill the
	// consistency and uniqueness requirements described in the Desc
	// documentation.
```

Reply:
We need at least a noop as the Prometheus interface requires it at runtime.